packages/ihe-gateway-sdk/src/client/ihe-gateway.ts

```diff
@@ -0,0 +1,43 @@
+import { Patient } from "@medplum/fhirtypes";
+import { v4 as uuidv4 } from "uuid";
+import { NPIStringArray, npiStringArraySchema } from "../models/shared";
+import { XCPDGateways, XCPDRequest, xcpdGatewaysSchema } from "../models/xcpd-request";
+
+export class IHEGateway {
+  /**
+   * An XCPD request interface
+   * @param patient The patient data in FHIR R4 format.
+   * @param cxId The customer ID.
+   * @param xcpdGateways The OIDs and XCPD ITI-55 URLs of each organization to make a request to.
+   * @param principalCareProviderNPIs The list of NPIs of the practitioners associated with the patient.
+   * @param requestId Optional. Unique ID for the request. If not provided, one will be created.
+   * @returns an XCPD request to be used with an IHE Gateway.
+   */
+  createXCPDRequest({
+    patient,
+    cxId,
+    xcpdGateways,
+    principalCareProviderNPIs,
+    requestId,
+  }: {
+    patient: Patient;
+    cxId: string;
+    xcpdGateways: XCPDGateways;
+    principalCareProviderNPIs?: NPIStringArray;
+    requestId?: string;
+  }): XCPDRequest {
+    const xcpdRequest: XCPDRequest = {
+      id: requestId ?? uuidv4(),
```

Comment:
nit: you were using uuidv7 and we changed as a test to see if it helped to solve the issue you were having with Sequelize.

If NABD, I'd revert to v7 - in case we store this on our DB we prefer v7 over v4 for DB performance.